.github/workflows/cd.yml

```diff
@@ -0,0 +1,57 @@
+name: cd
+on:
+  workflow_dispatch:
+  check_run:
+    types:
+      - completed
+
```

Comment:
Given the amount of different actions involved here and the fact we are passing the GITHUB_TOKEN we should drop as much privileges as possible. For example there is no need for this action to be able to write inside the repository. The only needed "write" call is probably for the update/creation of GH releases.



https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token